README.md

```diff
@@ -145,12 +138,80 @@ for a link-local address is auto-detected.
 
 ## Detailed Description of Capabilities
 
-Here are some more detailes on the additional capabilities of mavlink-router.
-
-### Endpoint Configuration
-
-A description of the different endpoint operating modes and their configuration
-parameters can be found in the [examples/config.sample](examples/config.sample).
+To understand how MAVLink Router forwards messages between endpoints, it
+important to know, that a MAVLink message has to have a sender address, but
+only some messages have a target address (e.g. parameter requests). The sender
+and target address consist of a system and component ID each, so one UAV can
+have a flight controller and companion computer with individual component IDs,
+but sharing the same system ID. In the target address, a component ID or 0 is
+used to broadcast to all components on a system, a sysID of 0 broadcasts to all
+systems.
+
+### Endpoints
+MAVLink Router supports three basic types of endpoints: UART, UDP link and TCP
+client. Additionally, it'll act as a TCP server for dynamic clients (if not
+explicitly deactivated).
+
+Endpoint types (see [examples/config.sample](examples/config.sample) for the
+config file format):
+
+  - UART: For telemetry radios or other serial links
+    * Configuration: UART device path/name and baudrate
+    * Behavior: Data is received and sent right away
+  - UDP:
+    * Configuration: Mode (client or server), IP address and port; both unicast
+      and multicast addresses are possible
+    * Behavior in client mode: Data is sent right away, but can only be
+      received on established socket connection
+    * Behavior in server mode: Data is received right away, but can only be
```

Comment:
I think the distinction is more on "is the port configure for sending or receiving?". Then when data can be received. Or we may want to be clearer on what "[sent|received| right away" means. Here it means "on program startup".